src/services/render/index.tsx

```diff
@@ -80,24 +75,27 @@ export const renderElement = (
     );
 
     if (Component) {
+      // Prepare props for the component
+      const props = {
+        element,
+        onUpdate,
+        options: {
+          ...options,
+          inlineFormattingContext,
+        },
+        stylesheets,
+      };
+
       /* eslint-disable max-len */
-      // Use object spreading instead of explicitly setting the key (to potentially undefined values)
-      // Explicitly setting the key causes collision when several components render with `undefined` value for `key`
-      // Object spreading will define the prop only when its value is truthy
-      /* eslint-enable max-len */
-      const extraProps: Partial<{ [key: string]: string | null }> = {};
-      if (element.getAttribute('key')) {
-        extraProps.key = element.getAttribute('key');
+      // Conditionally render the component with a key if it exists, to avoid
+      // warnings with current React versions, when the key attribute is set using the spread operator.
+      const key = element.getAttribute('key');
+
+      /* eslint-disable react/jsx-props-no-spreading */
```

Comment:
minor: either re-enable that rule after line 97, or explicitly disable both lines with `eslint-disable-line` comment.